cdk/domino_cdk/provisioners/vpc.py

```diff
@@ -209,7 +209,7 @@ def provision_bastion(self, name: str, bastion: config.VPC.Bastion) -> Optional[
                         40,  # note: this requires the AMI root device already be <=40GB
                         delete_on_termination=True,
                         encrypted=True,
-                        volume_type=ec2.EbsDeviceVolumeType.GP2,
+                        volume_type=ec2.EbsDeviceVolumeType.GP3,
```

Comment:
Curious what the upgrade experience is like for the bastion.

Reply:
We do not do re-provisioning, correct?
I.e. bastion is now provisioned with gp3 block storage and this is it.
What upgrade experience differences you expect?

Reply:
We don't reprovision as a matter of course on upgrades, but it's not true that we'll never ever reprovision.

The ideal outcome is that since the bastion is already provisioned, cloudformation doesn't bother reprovisioning the bastion despite the change since it already exists (and we don't care much about the bastion and its disk type). I don't expect this to happen, and wouldn't want to waste any time engineering it to happen.

What I assume will happen is that CloudFormation will destroy the bastion and create a new one. This is fine, as people really shouldn't store important stuff on the bastion and there will be probably minimal people that go through this upgrade over time. But it'd be nice to confirm this happens as expect, that we don't run into unexpected errors and can still connect to the deployment afterward (ie output w/the bastion ip address isn't cached on the deployer side or something).

Reply:
OK.let me run an experiment like this

Reply:
Switching the code from GP3 back to GP2 did not trigger reprovisioning of Bastion.

I'll try GP2-GP3 path

Reply:
Yes, CF scraps the old bastion instance and deploys a new one